pkg/webhooks/clusterlogging/clusterlogging.go

```diff
@@ -0,0 +1,250 @@
+}
+
+// Rules implements Webhook interface
+func (s *ClusterloggingWebhook) Rules() []admissionregv1.RuleWithOperations { return rules }
+
+// GetURI implements Webhook interface
+func (s *ClusterloggingWebhook) GetURI() string { return "/" + WebhookName }
+
+// SideEffects implements Webhook interface
+func (s *ClusterloggingWebhook) SideEffects() admissionregv1.SideEffectClass {
+	return admissionregv1.SideEffectClassNone
+}
+
+// Validate implements Webhook interface
+func (s *ClusterloggingWebhook) Validate(req admissionctl.Request) bool {
+	valid := true
+	valid = valid && (req.UserInfo.Username != "")
+	valid = valid && (req.Kind.Kind == ClusterLoggingKind)
+
+	return valid
+}
+
+type retentionPolicyValidator struct {
+	name       string
+	hint       string
+	upperBound esapi.TimeUnit
+}
+
+func (r *retentionPolicyValidator) checkPolicy(retentionPolicy *cl.RetentionPolicySpec) (bool, admissionctl.Response) {
+	isAllowed, deniedMessage, err := r.isAllowed(retentionPolicy)
+	if err != nil {
+		return false, admissionctl.Errored(http.StatusBadRequest, err)
+	}
+	if !isAllowed {
+		return false, admissionctl.Denied(deniedMessage)
+	}
+	return true, admissionctl.Allowed("Allowed to create ClusterLogging")
+}
+
+func (r *retentionPolicyValidator) isAllowed(retentionPolicy *cl.RetentionPolicySpec) (bool, string, error) {
+	if retentionPolicy == nil {
+		return false, "The entered retention policy is not allowed. " + r.name + " must not be unset. Hint: " + r.hint, nil
+	}
+
+	isAllowedRetentionDays, err := le(retentionPolicy.MaxAge, r.upperBound)
+	if err != nil {
+		log.Error(err, "Couldn't compare timeunits")
+		return false, "", err
+	}
+
+	if !isAllowedRetentionDays {
+		return false, "The entered RetentionPolicy " + r.name + " is not allowed. " + r.hint, nil
+	}
+
+	return true, "", nil
+}
+
+func (s *ClusterloggingWebhook) Authorized(request admissionctl.Request) admissionctl.Response {
+	r := s.authorized(request)
+	r.UID = request.AdmissionRequest.UID
+	return r
+}
+
+// Authorized implements Webhook interface
+func (s *ClusterloggingWebhook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+
+	decoder, err := admissionctl.NewDecoder(&s.s)
+	if err != nil {
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+	clusterLogging := &cl.ClusterLogging{}
+	if len(request.OldObject.Raw) > 0 {
+		err = decoder.DecodeRaw(request.OldObject, clusterLogging)
+	} else {
+		err = decoder.DecodeRaw(request.Object, clusterLogging)
+	}
+
+	if request.Operation == admissionapi.Delete {
+		return admissionctl.Allowed("Allowed to delete ClusterLogging")
+	}
+
+	if err != nil {
+		log.Error(err, "Couldn't decode ClusterLogging CR from incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+
+	retentionPolicy := clusterLogging.Spec.LogStore.RetentionPolicy
+
+	appValidator := retentionPolicyValidator{"app", "Set MaxAge to a value <= 7d", esapi.TimeUnit("7d")}
```

Comment:
```suggestion
	appValidator := retentionPolicyValidator{
		name:       "app",
		hint:       "Set MaxAge to a value <= 7d",
		upperBound: esapi.TimeUnit("7d"),
	}
```

To follow the style elsewhere in the project, please use this longer form struct declaration. The reasoning is that this long form helps readers follow along with the field, and also aids in searching the source file (such as for `hint`, as I did, to understand its purpose).

Alternately, this can be on one line if you deem that to be more readable:

```
	appValidator := retentionPolicyValidator{name: "app", hint: "Set MaxAge to a value <= 7d", upperBound: esapi.TimeUnit("7d")}
```

The auto-GitHub suggestion will apply the multi-line variant.